### PR TITLE
[FLINK-7835][cep] Fix duplicate() in NFASerializer.

### PR DESCRIPTION
**Pull request by kl0u**
R @aljoscha

### CODE REVIEW

File: flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/NonDuplicatingTypeSerializer.java [hunk 1]
```diff
@@ -190,13 +190,6 @@ public int hashCode() {
 		return typeSerializer.hashCode();
 	}
 
-	private void readObject(ObjectInputStream ois) throws IOException, ClassNotFoundException {
```
Comment (aljoscha) [inline flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/NonDuplicatingTypeSerializer.java]: removing this means that `elementList` and `identityMap` will not be initialised after deserialisation.

**Comment by kl0u** [comment]
Merging this.

**Comment by shashank734** [comment]
Can I directly use these changes in the release-1.3.2 tag or there in between dependencies?

**Comment by kl0u** [comment]
I think you can but you will have to recompile flink.

**Comment by shashank734** [comment]
While applying the patch it's giving a lot of conflicts. I think due to changes in between. So why I had doubts. I'll apply these changes manually.

**Comment by kl0u** [comment]
Yes, I think manually is the safest bet. 
Either way the 1.4 will come soon and this fix is going to be included.

**Comment by shashank734** [comment]
Yeah, But I think 1.4 will take some more couple of weeks. As unable to restore the state, So it's necessary to change for me.

**Comment by shashank734** [comment]
I think I have to wait for 1.4 cause there are a lot of changes in other files too between 1.3.2 release and this commit.

**Comment by kl0u** [comment]
Sorry to hear that @shashank734 . But on the bright side, the RC0 for Flink 1.4 is out:
http://apache-flink-mailing-list-archive.1008284.n3.nabble.com/PREVIEW-Release-1-4-0-release-candidate-0-td20237.html

So you can try it and also help us test it.

**Comment by shashank734** [comment]
Great

**Pull request merged**